### PR TITLE
school-library: Associations

### PR DESCRIPTION
**Pull request by Ceemos96**
### In this pull Request, I :

- Created a class Classroom with the following:
  
  - @Label instance variable, initialized in the constructor.
  
  - Setter and getter for @Label
  
- Created the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
 
  - Created the has-many side (a classroom has many students).
  
  - Created the belongs-to side (a student belongs to a classroom).
  
  - Made sure that when adding a student to a classroom it also sets the classroom for the student.
  
  - Made sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Created a class Book with the following:
   - @title and @author instance variables, initialized in the constructor.
   - Setters and getters for instance variables (remember about attr_accessor).
- Created a class Rental with the following:
   - @Date instance variable, initialized in the constructor.
   - Setter and getter for @Date .
- Created the many-to-many relationship between Person and Book using the intermediate class Rental. And implemented the following:
   - Created the has-many side of Book and Rental (a book has many rentals).
   - Created the belongs-to side of Rental and Book (a rental belongs to a book).
   - Created the has-many side of Person and Rental (a person has many rentals).
   - Created the belongs-to side of Rental and Person (a rental belongs to a person).
   - Modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

**Review by CollinsTatang — COMMENTED** [review]
Hi @Ceemos96 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have all the README file main header present.

- [ ] Please make sure all main headers from the [README template](https://github.com/microverseinc/readme-template) should be present.
- [ ] Please make sure that you add the section **Future Features**, including 1-3 potential features to be added.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: student.rb [hunk 1]
```diff
@@ -4,9 +4,14 @@ class Student < Person
...
   end
 
   def play_hooky
     '¯\(ツ)/¯'
   end
+
+  def add_classroom(classroom)
+    classroom.students.push(self) unless classroom.students.include?(self)
+  end
 end
```
Comment (Nemwel-Boniface) [inline student.rb:19]: - [ ] You might have forgotten to add the attribute reader for a classroom. Kindly include it ```attr_reader :classroom```.

File: person.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@
 require_relative 'trimmer_decorator'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (Nemwel-Boniface) [inline person.rb]: - [ ] You are required to set both the getter and setter for the rental, and the :rental is rather supposed to be with the attribute accessor and not the attribute reader. Kindly make this change. ```attr_accessor :name, :age, :rentals```.

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @Ceemos96      :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (AnselemOdims) [inline person.rb]: - [ ] As pointed out by the previous reviewer, please kindly include the `:rental` in the `attr_accessor`. This is to ensure that it has getter and a setter

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @Ceemos96 , 
Good job implementing some of the changes requested by the previous reviewer.

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Gopxfs — APPROVED** [review]
# 🟢 STATUS: APPROVED 🟢

## Hello @Ceemos96 👋,

Your project is complete! You followed all the requirements, feel free to merge it! 😄
Congratulations! 🎉 



## Highlights
✔️ Linters are passing successfully.
✔️ Implemented all the changes requested by the previous reviewer. Well done!

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**